### PR TITLE
ci: run instrumented tests on a hardware-accelerated emulator

### PR DESCRIPTION
**Pull request by Vladyslav-Soldatenko**
## Description

Adds a CI stage that actually boots an Android emulator and runs the existing `androidTest` suite (Compose UI tests, E2E flows, Room migration test) on every push/PR to `master`. Previously only `testDebugUnitTest` (JVM) ran in CI — the instrumented tests existed in the repo but nothing executed them automatically.

Along the way this also fixed a real bug in the E2E test suite's onboarding-dismissal helper, uncovered only because these tests had never actually run against a fresh install before.

## Type of Change

- [ ] New feature
- [ ] Bug fix
- [ ] Refactoring (no functional changes)
- [ ] Documentation update
- [x] Infrastructure / CI/CD
- [ ] Performance improvement

## Changes Made

- Add an `instrumented-tests` job to `.github/workflows/ci.yml`, gated on `needs: verify` so a lint/detekt/unit-test failure fails fast without spinning up an emulator.
- Use `reactivecircus/android-emulator-runner@v2` on `ubuntu-latest` with KVM enabled via udev rules, giving hardware-accelerated (not software-rendered) emulation.
- Target API 30 (matches `minSdk`), `default` image (no Google Play Services dependency in the app), `x86_64` arch, `pixel_6` hardware profile (1080x2400 — a realistic phone screen, not the tiny 320x640 default).
- Each run creates and boots a fresh AVD in one step — **deliberately no cross-run AVD cache**. An earlier attempt at caching (to save ~1 min of boot time) caused a run to hang for 4+ hours booting from a stale cross-run snapshot; the AVD boot isn't the bottleneck (37 sequential test-suite Activity launches are), so it wasn't worth the reliability risk.
- `timeout-minutes: 30` as a hard backstop so a future hang fails loudly instead of running unattended.
- Capture `adb logcat` for the duration of the run, uploaded as an artifact only `if: failure()` — cheap, and was the key diagnostic tool for the bugs below.
- Fixed `dismissOnboardingIfPresent` in `E2eTestSupport.kt`: on a real fresh install (no accessibility service enabled, no overlay permission granted), `MainActivity` shows two full-screen/dialog gating screens before the app's normal content. The "Not now" button on the first (a full-screen scrollable disclosure page, not a compact dialog) could be below the fold on the emulator's viewport — present in the semantics tree so the test's node lookup found it, but unreachable by `performClick()` without scrolling first. Added `performScrollTo()` before the click, falling back to a direct click when there's no scrollable ancestor (the second gating screen is a plain `AlertDialog`, nothing to scroll).
- Added an `OnboardingState.dismissed` short-circuit so the dismissal check only runs once per test-instrumentation process instead of every test re-polling for onboarding screens that are provably no longer there — cut the suite from ~13 minutes to ~4.5 minutes.

## How to Test

1. Open this PR and watch the `instrumented-tests` job in the Actions tab.
2. Confirm it depends on `verify` completing first, boots the emulator, and runs the full `androidTest` suite (37 tests: `DojoE2eTest`, `SettingsAnkiImportE2eTest`, `AppDatabaseMigrationTest`, etc.) to green.
3. On failure, confirm the logcat artifact uploads and contains useful diagnostic output.

## Checklist

- [x] Code follows the project's style guidelines
- [x] Self-reviewed the code for obvious errors
- [x] Added or updated tests where applicable, including edge cases
- [x] Existing tests pass locally
- [ ] Updated documentation if needed
- [x] No duplication introduced. I searched the existing code for similar behavior and extracted similar parts to be reused
- [x] No new warnings or supressions introduced. If something like this has to be introduced - get approve from maintainer first

## Related Issues

N/A — follow-up to setting up instrumented test coverage; this wires it into CI.

### CODE REVIEW

File: app/src/androidTest/java/com/procrastilearn/app/e2e/E2eTestSupport.kt [hunk 1]
```diff
@@ -8,9 +8,15 @@ import androidx.compose.ui.test.junit4.ComposeTestRule
...
 import androidx.compose.ui.test.performClick
+import androidx.compose.ui.test.performScrollTo
 import com.procrastilearn.app.R
 
-private const val ONBOARDING_STEP_TIMEOUT_MS = 1_000L
+// Generous enough to cover MainActivity's cold-start (Hilt injection, DataStore reads, first
+// Compose frame) on slow CI emulators — nodeVisibleWithin returns as soon as the node appears,
+// so this doesn't slow down fast environments, only raises the worst-case ceiling. A tighter
+// budget risks one gating screen's slow first render eating the whole repeat() iteration,
+// leaving the next gating screen (e.g. the overlay permission dialog) never dismissed.
```
Comment (Vladyslav-Soldatenko) [inline app/src/androidTest/java/com/procrastilearn/app/e2e/E2eTestSupport.kt]: remove comment
